hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/ethereum/CompositeEthereumTransactionParser.java

```
@@ -0,0 +1,59 @@
+package com.hedera.mirror.importer.parser.record.ethereum;
+
+/*-
+ * ‌
+ * Hedera Mirror Node
+ * ​
+ * Copyright (C) 2019 - 2022 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import javax.inject.Named;
+import lombok.RequiredArgsConstructor;
+import org.springframework.context.annotation.Primary;
+
+import com.hedera.mirror.common.domain.transaction.EthereumTransaction;
+import com.hedera.mirror.importer.exception.InvalidDatasetException;
+
+@Named
+@Primary
+@RequiredArgsConstructor
+public class CompositeEthereumTransactionParser implements EthereumTransactionParser {
+    private final LegacyEthereumTransactionParser legacyEthereumTransactionParser;
+    private final Eip1559EthereumTransactionParser eip1559EthereumTransactionParser;
+
+    @Override
+    public EthereumTransaction decode(byte[] transactionBytes) {
+        var ethereumTransactionParser = getEthereumTransactionParser(transactionBytes);
+        var ethereumTransaction = ethereumTransactionParser.decode(transactionBytes);
+        if (ethereumTransaction == null) {
+            throw new InvalidDatasetException("Unable to decode ethereum transaction bytes");
```

The throw should be pushed down to the concrete parsers instead of them returning null so it can return accurate information as to why it was unable to decode.

---

Pushed down